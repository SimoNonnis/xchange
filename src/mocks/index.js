const GBP = 'GBP';
const EUR = 'EUR';
const USD = 'USD';
const RUB = 'RUB';

export const pocketsInfo = {
  [GBP]: {
    code: GBP,
    name: 'British Pound',
    symbol: '£'
  },
  [EUR]: {
    code: EUR,
    name: 'Euro',
    symbol: '€'
  },
  [USD]: {
    code: USD,
    name: 'American Dollar',
    symbol: '$'
  },
  [RUB]: {
    code: RUB,
    name: 'Russian Ruble',
    symbol: '₽'
  }
};

export const pocketsAmount = {
  [GBP]: {
    amount: 1000
  },
  [EUR]: {
    amount: 1500
  },
  [USD]: {
    amount: 2000
  },
  [RUB]: {
    amount: 2500
  }
};

export const currenciesList = [GBP, EUR, USD];

export const pocketsList = [GBP, EUR, USD, RUB];

export const selected = RUB;

export const rates = {
  [GBP]: 1000,
  [EUR]: 2000,
  [USD]: 3000
};
