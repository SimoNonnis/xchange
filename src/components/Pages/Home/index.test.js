import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { pocketsList, pocketsInfo, pocketsAmount, rates } from '../../../mocks';
import { Home } from './index';

describe('Test <Home />', () => {
  it('should render with required props', () => {
    const wrapper = shallow(
      <Home
        pocketsList={pocketsList}
        pocketsInfo={pocketsInfo}
        pocketsAmount={pocketsAmount}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call all action with isPolling = true', () => {
    const wrapper = shallow(
      <Home
        pocketsList={pocketsList}
        pocketsInfo={pocketsInfo}
        pocketsAmount={pocketsAmount}
        isPolling={true}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
