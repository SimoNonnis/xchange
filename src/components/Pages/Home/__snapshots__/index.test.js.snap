// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Home /> should call all action with isPolling = true 1`] = `
<div>
  <h1
    className="pageTitle"
  >
    Pockets
  </h1>
  <div>
    <PocketButton
      amount={1000}
      className="buttonPocket"
      code="GBP"
      isSelected={false}
      key="GBP"
      name="British Pound"
      onClick={[Function]}
      symbol="£"
    />
    <PocketButton
      amount={1500}
      className="buttonPocket"
      code="EUR"
      isSelected={false}
      key="EUR"
      name="Euro"
      onClick={[Function]}
      symbol="€"
    />
    <PocketButton
      amount={2000}
      className="buttonPocket"
      code="USD"
      isSelected={false}
      key="USD"
      name="American Dollar"
      onClick={[Function]}
      symbol="$"
    />
    <PocketButton
      amount={2500}
      className="buttonPocket"
      code="RUB"
      isSelected={false}
      key="RUB"
      name="Russian Ruble"
      onClick={[Function]}
      symbol="₽"
    />
  </div>
  <div
    className="u-topMargin"
  >
    <span>
      Select your pocket ↑
    </span>
  </div>
</div>
`;

exports[`Test <Home /> should render with required props 1`] = `
<div>
  <h1
    className="pageTitle"
  >
    Pockets
  </h1>
  <div>
    <PocketButton
      amount={1000}
      className="buttonPocket"
      code="GBP"
      isSelected={false}
      key="GBP"
      name="British Pound"
      onClick={[Function]}
      symbol="£"
    />
    <PocketButton
      amount={1500}
      className="buttonPocket"
      code="EUR"
      isSelected={false}
      key="EUR"
      name="Euro"
      onClick={[Function]}
      symbol="€"
    />
    <PocketButton
      amount={2000}
      className="buttonPocket"
      code="USD"
      isSelected={false}
      key="USD"
      name="American Dollar"
      onClick={[Function]}
      symbol="$"
    />
    <PocketButton
      amount={2500}
      className="buttonPocket"
      code="RUB"
      isSelected={false}
      key="RUB"
      name="Russian Ruble"
      onClick={[Function]}
      symbol="₽"
    />
  </div>
  <div
    className="u-topMargin"
  >
    <span>
      Select your pocket ↑
    </span>
  </div>
</div>
`;
